internal/internal_workflow.go

```diff
@@ -972,20 +974,53 @@ func getStackTrace(coroutineName, status string, stackDepth int) string {
 
 func getStackTraceRaw(top string, omitTop, omitBottom int) string {
 	stack := stackBuf[:runtime.Stack(stackBuf[:], false)]
-	rawStack := strings.TrimRightFunc(string(stack), unicode.IsSpace)
+	outStack := filterStackTrace(string(stack), omitTop, omitBottom)
+	return strings.Join([]string{top, outStack}, "\n")
+}
+
+func filterStackTrace(stack string, omitTop, omitBottom int) string {
+	stack = strings.TrimRightFunc(stack, unicode.IsSpace)
 	if disableCleanStackTraces {
-		return rawStack
+		return stack
 	}
-	lines := strings.Split(rawStack, "\n")
+
+	lines := strings.Split(stack, "\n")
 	omitEnd := len(lines) - omitBottom
 	// If the start is after the end, the depth was invalid originally so return
 	// the entire raw stack
 	if omitTop > omitEnd {
-		return rawStack
+		return stack
+	}
+	return strings.Join(lines[omitTop:omitEnd], "\n")
+}
+
+func getCoroStackTrace(crt *coroutineState, status string, stackDepth int) string {
```

Comment:
Not a huge fan of this approach of using the pointer to `coroutineState` as the identifying ID since Go Lang is free to move the pointer afaik. I would prefer using a generated UUID approach personally.

Reply:
If Go moved the `coroutineState`, it would have to update all pointers to it. That is, both `crt` here and the value in the stack trace would change. However, I do see the (small) possibility of the pointer changing between the stack trace dump and the formatting of `needle`.

A different identifier needs to be stored both inside `coroutineState` and passed as an argument to `s.run(...)`. The stack trace truncates UUIDs passed as function arguments, so it would have to be something smaller. Right now, the stack trace code reads 8 bytes of each argument, so we could use a monotonic, wrapping `uint64` counter.

I personally like the pointer approach because its completely free until a stack trace is taken, but I'm also open to implementing a different approach (like the one above). Let me know what you prefer.

Reply:
I'm fine leaving it using the pointer for now. If we get lots of reports of "<coroutine stack not found>" then we can investigate changing the implementation.

Reply:
I added a comment to the relevant error-return in case the race condition does start to occur. Go's GC is currently non-moving, so it shouldn't for the time being: https://tip.golang.org/doc/gc-guide#Tracing_Garbage_Collection